Sources/Basics/SandboxProfile.swift

```diff
@@ -0,0 +1,103 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 - 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+import TSCBasic
+
+/// A sandbox profile representing the desired restrictions. The implementation can vary between platforms.
+public struct SandboxProfile {
+    /// An ordered list of path rules, where the last rule to cover a particular path "wins". These will be resolved
+    /// to absolute paths at the time the profile is applied. They are applied after any of the implicit directories
+    /// referenced by other sandbox profile settings.
+    public var pathAccessRules: [PathAccessRule]
+
+    /// Represents a rule for access to a path and everything under it.
+    public enum PathAccessRule: Equatable {
+        case noaccess(AbsolutePath)
+        case readonly(AbsolutePath)
+        case writable(AbsolutePath)
+    }
+
+    /// Whether to allow outbound and inbound network access.
+    public var allowNetwork: Bool
```

Comment:
not directly related to this PR, but we may need to make this more robust to allow some level of network access, for example with localhost on specific port

Reply:
Yes, that's a good point.  I wanted to add the flag here to indicate that this is not just about paths — I know you said not directly related to this PR, but do you feel we should hold off on this flag until then?

Reply:
FWIW, I removed it from this PR and just added a comment saying that we want this struct to have that kind of control.